packages/codegen.go/src/operations.ts

```diff
@@ -703,6 +703,10 @@ function createProtocolRequest(client: go.Client, method: go.Method | go.NextPag
     }
   }
   for (const param of headerParams.sort((a: go.HeaderParameter, b: go.HeaderParameter) => { return helpers.sortAscending(a.headerName, b.headerName);})) {
+    if (param.headerName.match(/^content-type$/)) {
+      // canonicalize content-type as req.SetBody checks for it via its canonicalized name :(
```

Comment:
For reference: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/internal/exported/request.go#L219

Reply:
My first thought was "we could change that comparison", but every change I came up with was worse than this

Reply:
That was my initial thought as well, and I came to the same conclusion.